src/main/java/org/graylog2/gelfclient/transport/AbstractGelfTransport.java

```diff
@@ -111,4 +117,16 @@ public boolean trySend(final GelfMessage message) {
     public void stop() {
         workerGroup.shutdownGracefully().syncUninterruptibly();
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void flushAndStopSynchronously(int waitDuration, TimeUnit timeUnit, int retries) {
+
+        if (senderThreadReference != null) {
```

Comment:
The `senderThreadReference` field itself will never be `null` because we set it in the constructor. I think we have to check if the reference value is null.

```suggestion
        if (senderThreadReference.get() != null) {
```